widgetssdk/src/main/java/com/glia/widgets/chat/ChatView.java

```diff
@@ -1230,22 +1232,28 @@ private void dismissAlertDialog() {
         }
     }
 
-    private static Uri chooseUriByRequestCode(int requestCode, Uri galeryImgUri, Uri cameraImgUri) {
-        if (requestCode == OPEN_DOCUMENT_ACTION_REQUEST) return galeryImgUri;
-        else if (requestCode == CAPTURE_IMAGE_ACTION_REQUEST) return cameraImgUri;
-        else return null;
-    }
-
     public void onActivityResult(int requestCode, int resultCode, Intent intent) {
-        if ((requestCode == OPEN_DOCUMENT_ACTION_REQUEST || requestCode == CAPTURE_IMAGE_ACTION_REQUEST || requestCode == CAPTURE_VIDEO_ACTION_REQUEST)
-                && resultCode == Activity.RESULT_OK) {
-            Uri dataUri = intent != null ? intent.getData() : null;
-            Uri uri = chooseUriByRequestCode(requestCode, dataUri, controller.getPhotoCaptureFileUri());
-            controller.setPhotoCaptureFileUri(null);
+
+        if (resultCode != Activity.RESULT_OK) return;
+
+        if (requestCode == OPEN_DOCUMENT_ACTION_REQUEST
```

Comment:
Worth separating to methods if you untie those here - gets pretty long otherwise